esp/esdllib/esdl_script.cpp

```diff
@@ -769,155 +1046,145 @@ class CEsdlCustomTransform : public CInterfaceOf<IEsdlCustomTransform>
 
-    if (content.length()!=0)
+    IEspContext *context = reinterpret_cast<IEspContext*>(scriptCtx->queryEspContext());
+
+    if (level >= LogMax)
     {
-        if (level >= LogMax)
-        {
-            DBGLOG("ORIGINAL content: %s", content.str());
-            StringBuffer marshalled;
-            if (bindingCfg)
-                toXML(bindingCfg, marshalled.clear());
-            DBGLOG("BINDING CONFIG: %s", marshalled.str());
-            if (tgtcfg)
-                toXML(tgtcfg, marshalled.clear());
-            DBGLOG("TARGET CONFIG: %s", marshalled.str());
-        }
+        StringBuffer logtxt;
+        scriptCtx->toXML(logtxt, srcSection, false);
+        DBGLOG("ORIGINAL content: %s", logtxt.str());
+        scriptCtx->toXML(logtxt.clear(), "config");
+        DBGLOG("BINDING CONFIG: %s", logtxt.str());
+        scriptCtx->toXML(logtxt.clear(), "target");
+        DBGLOG("TARGET CONFIG: %s", logtxt.str());
+    }
 
-        bool strictParams = bindingCfg ? bindingCfg->getPropBool("@strictParams", false) : false;
-        Owned<IXpathContext> xpathContext = getXpathContext(content.str(), strictParams);
+    bool strictParams = scriptCtx->getXPathBool("config/*/@strictParams", false);
+    Owned<IXpathContext> xpathContext = scriptCtx->createXpathContext(srcSection, strictParams);
 
-        StringArray prefixes;
-        for ( IEsdlCustomTransform * const & item : transforms)
-        {
-            if (item)
-                item->appendEsdlURIPrefixes(prefixes);
-        }
+    StringArray prefixes;
+    for ( IEsdlCustomTransform * const & item : transforms)
+    {
+        if (item)
+            item->appendEsdlURIPrefixes(prefixes);
+    }
 
-        registerEsdlXPathExtensions(xpathContext, context, prefixes);
+    registerEsdlXPathExtensions(xpathContext, scriptCtx, prefixes);
 
-        VStringBuffer ver("%g", context->getClientVersion());
-        if(!xpathContext->addVariable("clientversion", ver.str()))
-            OERRLOG("Could not set ESDL Script variable: clientversion:'%s'", ver.str());
+    VStringBuffer ver("%g", context->getClientVersion());
+    if(!xpathContext->addVariable("clientversion", ver.str()))
+        OERRLOG("Could not set ESDL Script variable: clientversion:'%s'", ver.str());
 
-        //in case transform wants to make use of these values:
-        //make them few well known values variables rather than inputs so they are automatically available
-        xpathContext->addVariable("query", tgtcfg->queryProp("@queryname"));
-        xpathContext->addVariable("method", method);
-        xpathContext->addVariable("service", service);
-        xpathContext->addVariable("request", reqtype);
+    //in case transform wants to make use of these values:
+    //make them few well known values variables rather than inputs so they are automatically available
+    StringBuffer temp;
+    xpathContext->addVariable("query", scriptCtx->getXPathString("target/*/@queryname", temp));
```

Comment:
Why not also move this statement for the 'query' variable into the IXpathContext* createXpathContext() function in libxml_xpathprocessor.cpp where the other similar statements moved?

Reply:
may move this in the future.  I'm just not sure where I want knowledge of the Target config section.  The other values are all from the ESDL section which comes from specific values in the ESDL definition.